ui/lib/kv/addon/routes/secret/details.js

```diff
@@ -24,10 +25,17 @@ export default class KvSecretDetailsRoute extends Route {
   setupController(controller, resolvedModel) {
     super.setupController(controller, resolvedModel);
 
-    controller.breadcrumbs = [
+    let breadcrumbsArray = [
       { label: 'secrets', route: 'secrets', linkExternal: true },
-      { label: resolvedModel.backend, route: 'secrets' },
-      { label: resolvedModel.path },
+      { label: resolvedModel.backend, route: 'list' },
     ];
+
+    if (pathIsFromDirectory(resolvedModel.path)) {
+      breadcrumbsArray = [...breadcrumbsArray, ...breadcrumbsForDirectory(resolvedModel.path)];
+    } else {
+      const breadcrumbsCurrentPath = { label: resolvedModel.path };
+      breadcrumbsArray.push(breadcrumbsCurrentPath);
```

Comment:
nit: this is a little easier for me to read, but no need to change - just a small suggestion
```suggestion
      breadcrumbsArray.push({ label: resolvedModel.path });
```